### PR TITLE
[DOC] Added missing documentation for constructors

### LINKED ISSUE #1638: [DOX] No documentation for the constructor of the LocalAlignmentEnumerator class
If you take a look, there is no documentation of the constructor:
http://docs.seqan.de/seqan/master/index.html?p=LocalAlignmentEnumerator

same here:
class Score http://docs.seqan.de/seqan/master/index.html?p=Score
class Index http://docs.seqan.de/seqan/master/index.html?p=Index

### PR DESCRIPTION
**Pull request by smehringer**
master PR #1776
Fixes #1638

### CODE REVIEW

File: include/seqan/align/local_alignment_enumeration.h [hunk 1]
```diff
@@ -63,6 +63,7 @@ class LocalAlignmentEnumerator;
  * @tparam TScore The type of the @link Score @endlink to use for the local alignment.
  * @tparam TSpec  The tag to use for specializing the enumerator.
  *
+ * This is an abstract class that will only be instantiated thorugh its subclasses.
```
Comment (rrahn) [inline include/seqan/align/local_alignment_enumeration.h]: typo: through

File: include/seqan/index/index_dfi.h [hunk 2]
```diff
@@ -215,6 +215,22 @@ namespace seqan
         TPredHull        predHull;
         TPred            pred;
 
+        /*!
+         * @fn IndexDfi::Index
+         * @brief Constructor
+         *
+         * @signature Index::Index();
+         * @signature Index::Index(index);
+         * @signature Index::Index(text);
```
Comment (rrahn) [inline include/seqan/index/index_dfi.h]: Make only one signature, with optional arguments like:
Index(text [, predHull] [,pred]) and write in the param section that this is an optional parameter.

File: include/seqan/index/index_fm.h [hunk 3]
```diff
@@ -205,6 +205,16 @@ class Index<TText, FMIndex<TSpec, TConfig> >
...
 
+    /*!
+     * @fn FMIndex::Index
+     * @brief Constructor
+     *
+     * @signature Index::Index();
+     * @signature Index::Index(text);
+     *
+     * @param[in] text The text to be indexed.
+     */
```
Comment (rrahn) [inline include/seqan/index/index_fm.h]: @h2 @cpockrandt  is it true, that the text must be in reversed for the FM-Index. If so we should drop a node in the description of the descriptor.

File: include/seqan/index/index_qgram.h [hunk 4]
```diff
@@ -257,6 +257,20 @@ class Index<TText_, IndexQGram<TShapeSpec, TSpec> > {
...
     TSize            stepSize;    // store every <stepSize>'th q-gram in the index
 
+    /*!
+     * @fn IndexQGram::Index
+     * @brief Constructor
+     *
+     * @signature Index::Index();
+     * @signature Index::Index(index);
+     * @signature Index::Index(text);
+     * @signature Index::Index(text, shape);
```
Comment (rrahn) [inline include/seqan/index/index_qgram.h]: Make shape as optional parameter in one signature.

File: include/seqan/index/index_qgram_openaddressing.h [hunk 5]
```diff
@@ -116,6 +116,21 @@ namespace seqan
...
 
+        /*!
+         * @fn OpenAddressingQGramIndex::Index
+         * @brief Constructor
+         *
+         * @signature Index::Index();
+         * @signature Index::Index(index);
+         * @signature Index::Index(text);
+         * @signature Index::Index(text, predHull);
+         * @signature Index::Index(text, predHull, pred);
```
Comment (rrahn) [inline include/seqan/index/index_qgram_openaddressing.h]: Copy'n'Paste error. shape instead predHull and pred

File: include/seqan/index/index_fm.h (hunk 3 above)
Comment (cpockrandt) [inline include/seqan/index/index_fm.h]: Yes, the text of the FM-index has to be reversed

Comment (smehringer) [inline include/seqan/index/index_fm.h]: so shall I change the documentation?
`Index::Index(revText)`
`revText   the reversed text to be indexed`

maybe with a link to the `reverse` function for strings?

Comment (cpockrandt) [inline include/seqan/index/index_fm.h]: Or we could refer to ModifiedStrings.

I think @h-2 uses the reverse function in Lambda. Maybe he can explain 
why he doesn't use a ModifiedString instead. Performance reasons?

Comment (h-2) [inline include/seqan/index/index_fm.h]: > If so, we should drop a node in the description of the descriptor.

Yes, please document that the all the talk about SATree and FibreSA is misleading, because the FM-Index in the current implementation does not provide a suffx trie interface, but a prefix trie interface. 

> Maybe he can explain why he doesn't use a ModifiedString instead. Performance reasons?

Yes, I recommond using `reverse()` because it moves the transformation to time-of-indexing, instead of time-of-access.

Comment (smehringer) [inline include/seqan/index/index_fm.h]: @cpockrandt will add a tag ("suffix") to the FMindex interface s.t. the user will be able to use the FMindex as a suffix-trie and the text does not need to be reversed beforehand. Since the documentation of this should be done in another pull request, the current documentation here is sufficient I think?
